Add Support for UCAT and UAAT Chunks

h2o-DS opened the pull request:
Some `NCER` files have `UCAT` chunks and some `NANR` files have `UAAT` chunks. This PR adds support for reading/writing these chunks.

h2o-DS committed:
Add UCAT and UAAT chunks

Add ability to read/write UCAT chunks in NCER files and UAAT chunks in NANR files

h2o-DS committed:
FrameAttributes

Stop recording cell attributes (all zero)
Properly record frame attributes
Calculate attribute pointers and chunk sizes

red031000 commented on gfx.c:
```diff
@@ -1025,6 +1027,18 @@ void ReadNtrCell_CEBK(unsigned char * restrict data, unsigned int blockOffset, u
+    {
+        offset = blockOffset + 0x18 + ucatOffset + 0x04 * options->cellCount;
+
+        options->ucatCellAttribtes = malloc(sizeof(int) * options->cellCount);
```
I'd personally prefer this one to be sizeof(uint32_t) or sizeof(int32_t)

red031000 replied:
ensure that malloc'd data is freed where appropriate
usually in the options free

red031000 commented on gfx.c:
```diff
@@ -1252,6 +1283,12 @@ void WriteNtrCell(char *path, struct JsonToCellOptions *options)
     }
 
+    // word-aligned
+    while (offset%4 > 0)
```
minor thing, space between % and both sides

red031000 commented on gfx.c:
```diff
@@ -1283,6 +1320,61 @@ void WriteNtrCell(char *path, struct JsonToCellOptions *options)
+    if (options->ucatEnabled)
+    {
+       // UCAT magic
+        KBECContents[offset] = 0x54;
```
I'd prefer these to be a strncpy with the string

red031000 commented on gfx.c:
```diff
@@ -1462,6 +1557,25 @@ void ReadNtrAnimation(char *path, struct JsonToAnimationOptions *options)
+    {
+        offset = 0x28 + uaatOffset + 0x0c * options->sequenceCount + 0x04 * options->frameCount; // index of first attribute
+
+        options->uaatData.sequenceAttributes = malloc(sizeof(int) * options->sequenceCount);
```
ucat feedback applies to uaat too

red031000 commented on json.c:
```diff
@@ -542,6 +568,30 @@ struct JsonToAnimationOptions *ParseNANRJson(char *path)
+        cJSON *uaatData = cJSON_GetObjectItemCaseSensitive(json, "uaatData");
+
+        cJSON *uaatSequences = cJSON_GetObjectItemCaseSensitive(uaatData, "sequenceAttributes");
+        options->uaatData.sequenceAttributes = malloc(sizeof(int) * options->sequenceCount);
```
sizeof int32_t

red031000 reviewed (changes requested):
I'll have to check on the format itself, this is manly for style

h2o-DS committed:
free in options

malloc with uint32_t instead of int
free data in proper free options functions
strcpy magic

red031000 reviewed (changes requested):
this does seem to be good, just two minor things
with them addressed I can merge this
I checked the format and it seems to be correct (as far as I can tell)

h2o-DS committed:
Merge branch 'master' into TACU

h2o-DS committed:
word-allign optimize
